### ISSUE TITLE
Add search by word (not starting with)

### ISSUE THREAD

**Issue by nmadruga** [sussol/universal-drug-codes-web-app#63]
First of all I think the app is looking good. The feature to search by code was very good for what I was doing - searching for medicine that I had with different name than it was registered there.
![image](https://user-images.githubusercontent.com/16461988/31158612-6923f800-a91f-11e7-8ad7-7fcaa565942f.png)

About the issue, the idea is to search for a word that could be not in the first one. That would be useful for API too. And to make it more efficient could just check on words starting with that and not find in the middle of words or could also start after 2 letters. What do you thing? 

For example: 
Ethinylestradiol  & levonorgestrel 30mcg & 150mcg tablet
I was trying to find by typing: levono, and couldn't find it.

**Comment by jtrein** [comment]
Hi @nmadruga, cheers for testing this! I noticed when I do a fuzzy search (box unchecked) it will find it. The "sensitivity" settings for that library's algorithm seem to be working.

What this does uncover is the fault in the UI, possibly. When unchecking the box, I never notify the user what the search would do. Maybe a tooltip on uncheck could solve this issue?

Thinking out loud about the opposite direction: I'm hesitant to add too many search buttons and options as the UI gets really bloated and even more difficult to use.

![screen shot 2017-10-04 at 09 12 48](https://user-images.githubusercontent.com/1144320/31158878-b1f37b54-a8e4-11e7-93b1-d6a52bbb52f4.png)

**Comment by jtrein** [comment]
To address the new search step you're proposing, I'm sure it could be done, but how close to fuzzy search are we getting at that point? I see the benefit, because technically this particular drug is inside the string and it does start with the criteria typed in.

**Comment by nmadruga** [comment]
@jtrein I think that solve the problem for me. And adding more buttons would indeed be complex for no reason.  I would suggest that the checkbox could start unselected to help with that, instead of having the tooltip to explain. Is there other reason for it start selected?
But only one thing that would still be a problem is in the API. Now I'm using it to show item universal codes in mSupply by the first 3 letters, but in that case it doesn't show the item because it isn't the first letter. Is it possible to search by every word using the API?

**Comment by jtrein** [comment]
Good news! I'm not sure who asked for it when they were here in Nepal, but that's why. I believe there was no issue for it.

Right, good point. Sometimes I couple the API too close to this particular web app in my mind. That can be done, I'm sure. Can you raise the issue in the API repository, and we can get started.

**Comment by jtrein** [comment]
I don't have an issue with it being unchecked. Can you clear that first with NZ peeps, to be safe?

**Comment by nmadruga** [comment]
@jtrein Yep, cool I can talk with guys here. And will create the issue on the other repo. This was a nice chat.

**Comment by nmadruga** [comment]
I talked with Craig and he thinks is more common use with the box checked, because people would expect to be searching only for the first word. So let's leave like that.

**Comment by jtrein** [comment]
OK, cool! So, to be clear. Do you want to change the behaviour of this particular app (according to the new option: https://github.com/sussol/universal-drug-codes-server/issues/34)?
